tools/process_snaps.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# coding: utf-8
+
+import json
+import logging
+import os
+import re
+
+from launchpadlib.credentials import (RequestTokenAuthorizationEngine,
+                                      UnencryptedFileCredentialStore)
+from launchpadlib.launchpad import Launchpad
+import requests
+
+
+logger = logging.getLogger("mir.process_snaps")
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+APPLICATION = "mir-ci"
+LAUNCHPAD = "production"
+RELEASE = "xenial"
+TEAM = "mir-team"
+SNAP = "mir-kiosk"
+SOURCE_NAME = "mir"
+CHANNELS = {
+    "edge": ("dev", "mir-kiosk-edge"),
+    "candidate": ("rc", "mir-kiosk-candidate"),
```

Comment:
Is this right?
I was thinking we'd use the channels like this:

     "edge": ("dev", "mir-kiosk-edge"),
     "beta": ("rc", "mir-kiosk-beta"),
     "candidate": ("release", "mir-kiosk-candidate"),

(We agree publishing to "stable" should remain a manual step.)

Reply:
I'm not sure, `rc` == `release candidate`, so wouldn't that be the right place? And `release` is just like `stable`, I wanted the snap existing in `candidate` to be released to `stable` at the same time packages get copied from `rc` to `release` - this way we'd release exactly the snap we tested.

I don't see a use for the `beta` channel at the moment.

Reply:
*Maybe* we can build `rc` packages to `beta` and then build a tagged release to `candidate` and then manually publish to `stable`... but why?

Reply:
That's also a reasonable workflow (and the names do support it). My preference is to automate the 1-1 mapping from PPA to snap "e.g. candidate is *always* what's in release".

Guess we need a tie-breaker?

Reply:
@gerboland WDYT?

Reply:
I'm with @Saviq here. I don't see point of using all 4 streams, or the manual step.

If later we see the manual step is needed, easy to use the beta channel and do the manual push. But I'd rather try full automation first.